packages/devtools_app/lib/src/charts/flutter/flame_chart.dart

```diff
@@ -60,50 +63,98 @@ abstract class FlameChartState<T extends FlameChart, V> extends State<T>
   // each node is positioned inside its own list.
   final flameChartNodeTop = 0.0;
 
-  LinkedScrollControllerGroup _linkedScrollControllerGroup;
-
   final List<FlameChartRow> rows = [];
 
   final List<FlameChartSection> sections = [];
 
+  ScrollController verticalScrollController;
+
+  LinkedScrollControllerGroup linkedHorizontalScrollControllerGroup;
+
+  double verticalScrollOffset = 0.0;
+
+  double horizontalScrollOffset = 0.0;
+
+  double zoom = minZoomLevel;
+
   /// Starting pixels per microsecond in order to fit all the data in view at
   /// start.
   double get startingPxPerMicro =>
       widget.startingContentWidth / widget.data.time.duration.inMicroseconds;
 
   int get startTimeOffset => widget.data.time.start.inMicroseconds;
 
+  /// Provides CustomPaint widgets to be painted on top of the flame chart, if
+  /// overridden.
+  ///
+  /// The painters will be painted in the order that they are returned.
+  List<CustomPaint> buildCustomPaints(BoxConstraints constraints) => [];
+
   @override
   void initState() {
     super.initState();
     initFlameChartElements();
-    _linkedScrollControllerGroup = LinkedScrollControllerGroup();
+
+    linkedHorizontalScrollControllerGroup = LinkedScrollControllerGroup();
+    linkedHorizontalScrollControllerGroup.onOffsetChanged(() {
```

Comment:
nit: you could use cascades for these two calls.

Reply:
done